src/Common/src/TypeSystem/Interop/MethodDesc.Interop.cs

```diff
@@ -63,11 +63,51 @@ public struct ParameterMetadata
         public bool HasFieldMarshal { get { return (_attributes & ParameterMetadataAttributes.HasFieldMarshal) == ParameterMetadataAttributes.HasFieldMarshal; } }
 
 
-        public ParameterMetadata(int index, ParameterMetadataAttributes attributes, TypeDesc marshalAs)
+        public ParameterMetadata(int index, ParameterMetadataAttributes attributes, MarshalAsDescriptor marshalAsDescriptor)
         {
             Index = index;
             _attributes = attributes;
-            MarshalAs = marshalAs;       
+            MarshalAsDescriptor = marshalAsDescriptor;
+        }
+    }
+
+    [Flags]
+    public enum NativeType : byte
+    {
+        NATIVE_TYPE_INVALID = 0x0,
```

Comment:
Nit: Can you C#-ize the names in this enum?

E.g. the `SignatureTypeCode` enum in System.Reflection.Metadata we use around here is a C#-ized equivalent of `CorElementType` in the CLR.